### PR TITLE
668 collapse cost charge

### PR DESCRIPTION
**Pull request by adberglund**
## Summary 
This adds infrastructure to what was previously the OpenShift `/charge/` endpoint and renames it to `/costs/`. What was the OpenShift `/costs/` endpoint is now `/infrastructures/aws/costs/` as it contains just the infrastructure costs for OpenShift on AWS and allows for grouping by AWS specific concepts (e.g. region).

## Testing and Examples

Realizing that I did not set a pricelist for my examples, so there's no derived cost. Can edit these tomorrow.

```
GET /local/v1/reports/openshift/costs/?filter[time_scope_value]=-1&filter[time_scope_units]=month&filter[resolution]=monthly
HTTP 200 OK
Allow: GET, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "meta": {
        "count": 1,
        "filter": {
            "resolution": "monthly",
            "time_scope_value": "-1",
            "time_scope_units": "month"
        },
        "total": {
            "infrastructure_cost": {
                "value": 7843.0,
                "units": "USD"
            },
            "derived_cost": {
                "value": 0.0,
                "units": "USD"
            },
            "cost": {
                "value": 7843.0,
                "units": "USD"
            }
        }
    },
    "links": {
        "first": "/local/v1/reports/openshift/costs/?filter%5Bresolution%5D=monthly&filter%5Btime_scope_units%5D=month&filter%5Btime_scope_value%5D=-1&limit=100&offset=0",
        "next": null,
        "previous": null,
        "last": "/local/v1/reports/openshift/costs/?filter%5Bresolution%5D=monthly&filter%5Btime_scope_units%5D=month&filter%5Btime_scope_value%5D=-1&limit=100&offset=0"
    },
    "data": [
        {
            "date": "2019-03",
            "values": [
                {
                    "date": "2019-03",
                    "infrastructure_cost": {
                        "value": 7843.0,
                        "units": "USD"
                    },
                    "derived_cost": {
                        "value": 0.0,
                        "units": "USD"
                    },
                    "cost": {
                        "value": 7843.0,
                        "units": "USD"
                    }
                }
            ]
        }
    ]
}
```

```
GET /local/v1/reports/openshift/costs/?filter[time_scope_value]=-1&filter[time_scope_units]=month&filter[resolution]=monthly&group_by[node]=*
HTTP 200 OK
Allow: GET, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "meta": {
        "count": 1,
        "group_by": {
            "node": [
                "*"
            ]
        },
        "filter": {
            "resolution": "monthly",
            "time_scope_value": "-1",
            "time_scope_units": "month"
        },
        "total": {
            "infrastructure_cost": {
                "value": 7843.0,
                "units": "USD"
            },
            "derived_cost": {
                "value": 0.0,
                "units": "USD"
            },
            "cost": {
                "value": 7843.0,
                "units": "USD"
            }
        }
    },
    "links": {
        "first": "/local/v1/reports/openshift/costs/?filter%5Bresolution%5D=monthly&filter%5Btime_scope_units%5D=month&filter%5Btime_scope_value%5D=-1&group_by%5Bnode%5D=%2A&limit=100&offset=0",
        "next": null,
        "previous": null,
        "last": "/local/v1/reports/openshift/costs/?filter%5Bresolution%5D=monthly&filter%5Btime_scope_units%5D=month&filter%5Btime_scope_value%5D=-1&group_by%5Bnode%5D=%2A&limit=100&offset=0"
    },
    "data": [
        {
            "date": "2019-03",
            "nodes": [
                {
                    "node": "alpha",
                    "values": [
                        {
                            "node": "alpha",
                            "date": "2019-03",
                            "infrastructure_cost": {
                                "value": 7843.0,
                                "units": "USD"
                            },
                            "derived_cost": {
                                "value": 0.0,
                                "units": "USD"
                            },
                            "cost": {
                                "value": 7843.0,
                                "units": "USD"
                            }
                        }
                    ]
                },
                {
                    "node": "bravo",
                    "values": [
                        {
                            "node": "bravo",
                            "date": "2019-03",
                            "infrastructure_cost": {
                                "value": 0.0,
                                "units": "USD"
                            },
                            "derived_cost": {
                                "value": 0.0,
                                "units": "USD"
                            },
                            "cost": {
                                "value": 0.0,
                                "units": "USD"
                            }
                        }
                    ]
                }
            ]
        }
    ]
}
```

```
GET /local/v1/reports/openshift/costs/?filter[time_scope_value]=-1&filter[time_scope_units]=month&filter[resolution]=monthly&group_by[cluster]=*
HTTP 200 OK
Allow: GET, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "meta": {
        "count": 1,
        "group_by": {
            "cluster": [
                "*"
            ]
        },
        "filter": {
            "resolution": "monthly",
            "time_scope_value": "-1",
            "time_scope_units": "month"
        },
        "total": {
            "infrastructure_cost": {
                "value": 7843.0,
                "units": "USD"
            },
            "derived_cost": {
                "value": 0.0,
                "units": "USD"
            },
            "cost": {
                "value": 7843.0,
                "units": "USD"
            }
        }
    },
    "links": {
        "first": "/local/v1/reports/openshift/costs/?filter%5Bresolution%5D=monthly&filter%5Btime_scope_units%5D=month&filter%5Btime_scope_value%5D=-1&group_by%5Bcluster%5D=%2A&limit=100&offset=0",
        "next": null,
        "previous": null,
        "last": "/local/v1/reports/openshift/costs/?filter%5Bresolution%5D=monthly&filter%5Btime_scope_units%5D=month&filter%5Btime_scope_value%5D=-1&group_by%5Bcluster%5D=%2A&limit=100&offset=0"
    },
    "data": [
        {
            "date": "2019-03",
            "clusters": [
                {
                    "cluster": "nise",
                    "values": [
                        {
                            "date": "2019-03",
                            "cluster": "nise",
                            "infrastructure_cost": {
                                "value": 7843.0,
                                "units": "USD"
                            },
                            "derived_cost": {
                                "value": 0.0,
                                "units": "USD"
                            },
                            "cost": {
                                "value": 7843.0,
                                "units": "USD"
                            },
                            "cluster_alias": "Test Provider"
                        }
                    ]
                }
            ]
        }
    ]
}
```

```
GET /local/v1/reports/openshift/costs/?filter[time_scope_value]=-1&filter[time_scope_units]=month&filter[resolution]=monthly&group_by[project]=*
HTTP 200 OK
Allow: GET, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "meta": {
        "count": 1,
        "group_by": {
            "project": [
                "*"
            ]
        },
        "filter": {
            "resolution": "monthly",
            "time_scope_value": "-1",
            "time_scope_units": "month"
        },
        "total": {
            "infrastructure_cost": {
                "value": 1960.75,
                "units": "USD"
            },
            "derived_cost": {
                "value": 0.0,
                "units": "USD"
            },
            "cost": {
                "value": 1960.75,
                "units": "USD"
            }
        }
    },
    "links": {
        "first": "/local/v1/reports/openshift/costs/?filter%5Bresolution%5D=monthly&filter%5Btime_scope_units%5D=month&filter%5Btime_scope_value%5D=-1&group_by%5Bproject%5D=%2A&limit=100&offset=0",
        "next": null,
        "previous": null,
        "last": "/local/v1/reports/openshift/costs/?filter%5Bresolution%5D=monthly&filter%5Btime_scope_units%5D=month&filter%5Btime_scope_value%5D=-1&group_by%5Bproject%5D=%2A&limit=100&offset=0"
    },
    "data": [
        {
            "date": "2019-03",
            "projects": [
                {
                    "project": "namespace_ci",
                    "values": [
                        {
                            "date": "2019-03",
                            "project": "namespace_ci",
                            "infrastructure_cost": {
                                "value": 1960.75,
                                "units": "USD"
                            },
                            "derived_cost": {
                                "value": 0.0,
                                "units": "USD"
                            },
                            "cost": {
                                "value": 1960.75,
                                "units": "USD"
                            }
                        }
                    ]
                },
                {
                    "project": "namespace_hyper",
                    "values": [
                        {
                            "date": "2019-03",
                            "project": "namespace_hyper",
                            "infrastructure_cost": {
                                "value": 0.0,
                                "units": "USD"
                            },
                            "derived_cost": {
                                "value": 0.0,
                                "units": "USD"
                            },
                            "cost": {
                                "value": 0.0,
                                "units": "USD"
                            }
                        }
                    ]
                }
            ]
        }
    ]
}
```

### CODE REVIEW

File: koku/api/report/ocp/view.py [hunk 1]
```diff
@@ -243,14 +243,14 @@ def cpu(request):
...
 @permission_classes([AllowAny])
 @renderer_classes(tuple(api_settings.DEFAULT_RENDERER_CLASSES))
-def charges(request):
-    """Get OpenShift charge data.
+def costs(request):
+    """Get OpenShift cost data.
 
-    @api {get} /cost-management/v1/reports/openshift/charge/ Get OpenShift charge data
-    @apiName getOpenShiftChargeData
+    @api {get} /cost-management/v1/reports/openshift/cost/ Get OpenShift costs data
```
Comment (chambridge) [inline koku/api/report/ocp/view.py]: `cost` --> `costs`

File: koku/api/report/ocp/view.py [hunk 2]
```diff
@@ -267,55 +267,97 @@ def charges(request):
...
+                        "units": "USD"
+                    }
+                }
             },
-            "filter": {
-                "resolution": "monthly",
-                "time_scope_value": "-1",
-                "time_scope_units": "month"
+            "links": {
+                "first": "/local/v1/reports/openshift/costs/?filter%5Bresolution%5D=monthly&filter%5Btime_scope_units%5D=month&filter%5Btime_scope_value%5D=-1&group_by%5Bproject%5D=%2A&limit=100&offset=0",  # noqa: E501
```
Comment (chambridge) [inline koku/api/report/ocp/view.py]: `/local/v1` --> `/cost-management/v1`

File: koku/api/report/ocp/view.py (hunk 1 above)
Comment (adberglund) [inline koku/api/report/ocp/view.py]:
```suggestion
    @api {get} /cost-management/v1/reports/openshift/costs/ Get OpenShift costs data
```

File: koku/api/report/ocp/view.py (hunk 2 above)
Comment (adberglund) [inline koku/api/report/ocp/view.py]:
```suggestion
                "first": "/cost-management/v1/reports/openshift/costs/?filter%5Bresolution%5D=monthly&filter%5Btime_scope_units%5D=month&filter%5Btime_scope_value%5D=-1&group_by%5Bproject%5D=%2A&limit=100&offset=0",  # noqa: E501
```

**Pull request merged**